Update to Sphinx-Gallery v0.1.11

Titan-C opened the pull request:
This update includes the style change of using `title` to describe code links. This change was left pending in https://github.com/scikit-learn/scikit-learn/pull/7986#issuecomment-272910678

It also make the gallery building robust on SyntaxErrors in the examples. Plus other enhancements not directly concerning Scikit-learn.

- Remains to figure out why in PR #7986 download boxed started to appear staked in wide windows displays and not in medium displays.

Titan-C commented:
Travis is failing because sphinx extensions are tested. Last version of the gallery imports from sphinx but sphinx is not installed for testing in travis. Shall sphinx be installed or shall this test be ignored?

lesteve commented:
AFAIK doc/sphintext are third party extensions, I don't think we should test them.

Maybe historically we had more custom code in doc/sphinxext, I am not sure.

jnothman commented:
I think the issue is more that historically we didn't have docs CI. So IMO
we can remove test-sphinxext, or add it to the Circle build.

On 24 January 2017 at 19:56, Loïc Estève <notifications@github.com> wrote:

> AFAIK doc/sphintext are third party extensions, I don't think we should
> test them.
>
> Maybe historically we had more custom code in doc/sphinxext, I am not sure.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/8222#issuecomment-274745190>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz67LK4A9zxn60ExTiTWEKkigCH6HZks5rVbzJgaJpZM4Lqft3>
> .
>

TomDLT commented:
#8228 is merged, if you update your PR with master, Travis should be green.

Titan-C commented:
Rebased, travis is green

Titan-C commented:
I figured it out about the download boxes being stacked. I comes from the responsive CSS layout and a mismatch in the width of the `.container` class from bootstrap. I could not make a better CSS, I just messed everything up. Thus my last commit gives the `.sphx-glr-footer .container` preference to have `width: auto`. This fixes the problem with the buttons being stacked and not in the center, without messing the complete CSS style of the scikit-learn website. It works well enough except for very wide screens, where the buttons are placed a bit more to the right.

lesteve commented:
I just pushed with `[doc build]` so we can look at the output in CircleCi.

lesteve commented:
@Titan-C I lost track of this one sorry. Is there anything else that needs to be done on the sphinx-gallery side?

Titan-C committed:
[doc build] Update to Sphinx-Gallery v0.1.11

Titan-C commented:
I'm bringing this back to life. Especially after the new Sphinx 1.6 release breaks the builds. There are some improvements from this version jump to the gallery 0.1.11. 
Relevant changes are as stated in the first comment. The SyntaxError capturing and the new tooltip style. The other changes are concerned with how Sphinx-Gallery handles paths internally, and some bug fixes.

lesteve commented:
Thanks for reviving this one. Not sure whether the CircleCI failure was a glitch, I kicked off a rebuild.

lesteve commented:
Hmmm the example error seems like a genuine problem with the ichart.yahoo.com website ... From a quick googling the service seems to have been discontinued by Yahoo. For example see this [link](https://forums.yahoo.net/t5/Yahoo-Finance-help/Is-Yahoo-Finance-API-broken/td-p/250503/page/3#messageview_4).

I guess we will need a work-around in the medium-term. 

@Titan-C for this PR you could add plot_stock_market.py  to the "expected to fail" examples.

Titan-C committed:
[doc build] example stock_marked is expected to fail

The service this example relies on seems discontinued, thus we skip it as
an error. Once this is fixed, Sphinx-Gallery will raise the errors that the
script is not failing anymore.

Titan-C committed:
[doc build] Make travis lint happy

lesteve commented on doc/conf.py:
```diff
@@ -236,12 +236,15 @@
         'numpy': 'http://docs.scipy.org/doc/numpy-1.6.0',
         'scipy': 'http://docs.scipy.org/doc/scipy-0.11.0/reference',
-        'nibabel': 'http://nipy.org/nibabel'}
+        'nibabel': 'http://nipy.org/nibabel'},
```
Not sure how nibabel got in there (probably from nilearn at one point) but we should remove it since nibabel is not used in scikit-learn examples.

naoyak commented:
Also related to Sphinx 1.6: #8828

Titan-C committed:
Drop nibabel for API parsing

lesteve commented:
The generated gallery is available [here](https://10348-843222-gh.circle-artifacts.com/0/home/ubuntu/scikit-learn/doc/_build/html/stable/auto_examples/index.html).

Can someone have a quick look to make sure there is nothing obviously wrong in the gallery and in the examples HTML? Maybe @jnothman @glemaitre 

I think this PR should be merged because:
* it's good to keep in sync with the latest sphinx-gallery
* it will keep PRs green until we figure out a work-around for https://github.com/scikit-learn/scikit-learn/issues/8899

glemaitre commented:
I'm taking it a look at it.

lesteve commented:
Quickly looking at it, the only thing I noticed is that the download buttons are shifted to the right. Compare [this PR](https://10348-843222-gh.circle-artifacts.com/0/home/ubuntu/scikit-learn/doc/_build/html/stable/auto_examples/plot_cv_predict.html#sphx-glr-auto-examples-plot-cv-predict-py) and [dev](http://scikit-learn.org/dev/auto_examples/plot_cv_predict.html#sphx-glr-auto-examples-plot-cv-predict-py).

@Titan-C are the download buttons supposed to be centered?

glemaitre commented:
@lesteve I could not spot anything wrong related to the update apart of what you mentioned.

However, I could spot a lot of examples with:

```python
####################
# some comments
```

It is usually used to split the code of the example. However, this is also the way the sphinx-gallery convert the example to notebook style. Therefore, some examples look a little bit messy. I would take some times to review the example and raise an issue.

lesteve committed:
Merge branch 'master' into glr_maint

lesteve commented:
I fixed the conflicts I introduced by merging #8921.

lesteve commented:
Let's merge this one!

lesteve commented:
Thanks @Titan-C!

lesteve commented:
> @Titan-C are the download buttons supposed to be centered?

@Titan-C do you have a comment about this?

Titan-C commented:
> @Titan-C are the download buttons supposed to be centered?

As I first wrote earlier in  https://github.com/scikit-learn/scikit-learn/pull/8222#issuecomment-275265527. I don't know why download buttons started to appear stacked, but in this PR I included a fix to give them  a nicer placement, but on big screens they look to be tilted to the right. Unfortunately to fix that you really need to mess with scikit-learn website layout a lot more.

lesteve commented:
OK thanks, I think this is fine for now.

jnothman commented:
We have been having Circle CI failures on master:
```
Exception occurred:
  File "/home/ubuntu/scikit-learn/doc/sphinxext/sphinx_gallery/gen_gallery.py", line 244, in setup
    if 'sphinx.ext.autodoc' in app._extensions:
AttributeError: 'Sphinx' object has no attribute '_extensions'

```

glemaitre commented:
The private function in sphinx has been renamed if I am not wrong. I saw
that somewhere

On 14 June 2017 at 14:10, Joel Nothman <notifications@github.com> wrote:

> We have been having Circle CI failures on master:
>
> Exception occurred:
>   File "/home/ubuntu/scikit-learn/doc/sphinxext/sphinx_gallery/gen_gallery.py", line 244, in setup
>     if 'sphinx.ext.autodoc' in app._extensions:
> AttributeError: 'Sphinx' object has no attribute '_extensions'
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/8222#issuecomment-308411734>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHG9PymPuvgtiyomRTMhRjb9fZtmOhG2ks5sD82ogaJpZM4Lqft3>
> .
>



-- 
Guillaume Lemaitre
INRIA Saclay - Parietal team
Center for Data Science Paris-Saclay
https://glemaitre.github.io/

Titan-C commented:
It is very strange that CI fails with that error. A fix to that is
pushed in
https://github.com/sphinx-gallery/sphinx-gallery/commit/e05e7d93a60240bfc98bf4e252a2152f38a3445a,
and is included in this PR in this change https://github.com/scikit-learn/scikit-learn/pull/8222/files#diff-cc293bc3f7189fb04db7447b6ffbc49fL244
Might it be that other merges to master have overwritten this change?

lesteve commented:
Hmmm I don't see any recent failures in https://circleci.com/gh/scikit-learn/scikit-learn.

jnothman commented:
Yes, I'm not sure why I've received emails.

On 14 June 2017 at 23:50, Loïc Estève <notifications@github.com> wrote:

> Hmmm I don't see any recent failures in https://circleci.com/gh/
> scikit-learn/scikit-learn.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/8222#issuecomment-308437138>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz610NTS9vTk3nsB-3scIJ3tcTlq9Xks5sD-UVgaJpZM4Lqft3>
> .
>
